x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/mapper/SemanticTextInferenceResultFieldMapperTests.java

```diff
@@ -0,0 +1,205 @@
+//
+//            assertNotNull(leaf.advance(0));
+//            assertEquals(0, leaf.doc());
+//            assertEquals(2, leaf.rootDoc());
+//            assertEquals(new SearchHit.NestedIdentity(fieldName1, 0, null), leaf.nestedIdentity());
+//
+//            assertNotNull(leaf.advance(1));
+//            assertEquals(1, leaf.doc());
+//            assertEquals(2, leaf.rootDoc());
+//            assertEquals(new SearchHit.NestedIdentity(fieldName2, 1, null), leaf.nestedIdentity());
+//
+//            assertNull(leaf.advance(2));
+//            assertEquals(2, leaf.doc());
+//            assertEquals(2, leaf.rootDoc());
+//            assertNull(leaf.nestedIdentity());
+//        });
+    }
+
+    private static void addSemanticTextMapping(XContentBuilder mappingBuilder, String fieldName, String modelId) throws IOException {
+        mappingBuilder.startObject(fieldName);
+        mappingBuilder.field("type", SemanticTextFieldMapper.CONTENT_TYPE);
+        mappingBuilder.field("model_id", modelId);
+        mappingBuilder.endObject();
+    }
+
+    private static SemanticTextInferenceResults generateSemanticTextinferenceResults(String semanticTextFieldName, List<String> chunks) {
+        List<SparseEmbeddingResults.Embedding> embeddings = new ArrayList<>(chunks.size());
+        for (String chunk : chunks) {
+            String[] tokens = chunk.split("\\s+");
+            List<SparseEmbeddingResults.WeightedToken> weightedTokens = Arrays.stream(tokens).map(
+                t -> new SparseEmbeddingResults.WeightedToken(t, randomFloat())).toList();
+
+            embeddings.add(new SparseEmbeddingResults.Embedding(weightedTokens, false));
+        }
+
+        return new SemanticTextInferenceResults(semanticTextFieldName, new SparseEmbeddingResults(embeddings), chunks);
+    }
+
+    private static void addSemanticTextInferenceResults(
+        XContentBuilder sourceBuilder,
+        List<SemanticTextInferenceResults> semanticTextInferenceResults) throws IOException {
+
+        Map<String, List<Map<String, Object>>> inferenceResultsMap = new HashMap<>();
+        for (SemanticTextInferenceResults semanticTextInferenceResult : semanticTextInferenceResults) {
+            List<Map<String, Object>> parsedInferenceResults = new ArrayList<>(semanticTextInferenceResult.text().size());
+
+            Iterator<SparseEmbeddingResults.Embedding> embeddingsIterator = semanticTextInferenceResult.sparseEmbeddingResults()
+                .embeddings().iterator();
+            Iterator<String> textIterator = semanticTextInferenceResult.text().iterator();
+            while (embeddingsIterator.hasNext() && textIterator.hasNext()) {
+                SparseEmbeddingResults.Embedding embedding = embeddingsIterator.next();
+                String text = textIterator.next();
+
+                parsedInferenceResults.add(Map.of(
+                    // TODO: OK to omit is_truncated?
```

Comment:
The `SparseEmbeddingResults.Embedding` class emits a map with an `is_truncated` key that i am omitting because it would break the expected format. It this OK? Is this a sign we need to adjust the expected format?

Reply:
Good point - let's check with David K on that

Reply:
Asked in [this slack thread](https://elastic.slack.com/archives/C0689SNEKPG/p1704484805359479)